Core/src/org/sleuthkit/autopsy/events/AutopsyEventPublisher.java

```diff
@@ -83,29 +120,22 @@ synchronized public void removeSubscriber(Collection<String> eventNames, Autopsy
      * @param eventNames The event the subscriber is no longer interested in.
      * @param subscriber The subscriber to remove.
      */
-    synchronized public void removeSubscriber(String eventName, AutopsyEventSubscriber subscriber) {
-        Set<AutopsyEventSubscriber> subscribers = subscribersByEvent.getOrDefault(eventName, null);
-        if (null != subscribers) {
-            subscribers.remove(subscriber);
-        }
+    synchronized public void removeSubscriber(String eventName, PropertyChangeListener subscriber) {
+        localPublisher.removeSubscriber(eventName, subscriber);
     }
 
     /**
-     * Publishes an event to all registered subscribers, even if a subscriber
-     * throws an exception.
+     * Publishes an event.
      *
-     * @param event The event to be published.
+     * @param event The event to publish.
+     * @throws JMSException
      */
-    synchronized public void publish(AutopsyEvent event) {
-        Set<AutopsyEventSubscriber> subscribers = subscribersByEvent.getOrDefault(event.getPropertyName(), null);
-        if (null != subscribers) {
-            for (AutopsyEventSubscriber subscriber : subscribers) {
-                try {
-                    subscriber.receiveEvent(event);
-                } catch (Exception ex) {
-                    logger.log(Level.SEVERE, "Exception thrown by subscriber", ex);
-                }
-            }
+    synchronized public void publish(AutopsyEvent event) throws JMSException {
+        event.setSourceType(AutopsyEvent.SourceType.LOCAL);
+        localPublisher.publish(event);
+        if (null != remotePublisher) {
```

Comment:
Okay. This makes sense. I had thought that it had already been published locally and we might be publishing it a second time, but that is not the case.